tests/dragonfly/replication_test.py

```diff
@@ -2411,3 +2411,33 @@ async def test_replicate_old_master(
     assert await c_replica.execute_command("get", "k1") == "v1"
 
     await disconnect_clients(c_master, c_replica)
+
+
+@pytest.mark.asyncio
+async def test_empty_hash_as_zipmap_bug(df_factory: DflyInstanceFactory):
```

Comment:
Can you kindly make sure that this test passes even if the code on the save path does not change?

Reply:
Can you plz elaborate what you mean here?

Reply:
This test won't pass without the fix since the replica will now contain an empty hashmap ?

Reply:
This PR attempts to fix both loading of an empty set in an RDB file (which shouldn't happen)
and also not saving an RDB file to begin with.

I wanted you to comment out the changes in `RdbSerializer::SaveEntry` so that we can confirm this works for bad RDB files?
I suspect that it won't, but I'd love to be surprised :)

Reply:
Yes the test won't pass without the changes in `SaveEntry()`

```
2440     # Key does not exist in replica because we skipped it                                                                                                                                                     
2441     assert await c_replica.execute_command(f"EXISTS foo") == 0 
```

Because we replicate an empty key. Now with the changes in `SaveEntry`, we no longer do that,

Reply:
Why would this test pass in the first place without the changes ?

Reply:
We want to write resilient code.
For example, if we have another bug - similar to this but other - we want to be able to load it.
So we need to support loading all kinds of data types which are empty (lists, hashes, etc), and handle them correctly without failing the load.

So yeah, we want to fix the save side (which you did, awesome), but also fix the load side to be more robust (that will likely require some more handling I'm afraif)

Reply:
dude, I was completely out of context :rofl: . When you said comment out `SaveEntry` I did not understand you wanted to test the `rdb loader` only so I blindly answered the test will fail because I assumed that the test would run `without any of the changes introduced by this PR). You are right we got a bug which is easy to handle and I will fix soon

Reply:
I was blindly got confused here. My fix was correct all along, in fact I introduced a full replication test that passes by loading an empty rdb file with the previous changes. 

I have no clue why the hell I got confused twice here.